extmod/moducryptolib.c

```diff
@@ -119,6 +144,33 @@ STATIC void aes_process_cbc_impl(AES_CTX_IMPL *ctx, const uint8_t *in, uint8_t *
         AES_cbc_decrypt(ctx, in, out, in_len);
     }
 }
+
+#if MICROPY_PY_UCRYPTOLIB_CTR
+// axTLS doesn't have CTR support out of the box. This is based on `mbedtls_aes_crypt_ctr`.
+STATIC void aes_process_ctr_impl(AES_CTX_IMPL *ctx, const uint8_t *in, uint8_t *out, size_t in_len, struct ctr_params *ctr_params) {
+    size_t n = ctr_params->nc_off;
+    uint8_t *const nonce_counter = ctx->iv;
+
+    while (in_len--) {
+        if (n == 0) {
+            aes_process_ecb_impl(ctx, nonce_counter, ctr_params->stream_block, true);
+
+            for (int i = 16; i > 0; i--) {
```

Comment:
Comment that this increments 128-bit counter?

Reply:
Okay